### ISSUE TITLE
Warning message in SSplotComparisons - brainstorm

### ISSUE THREAD

**Issue by kellijohnson-NOAA** [r4ss/r4ss#371]
https://github.com/r4ss/r4ss/blob/331061ff3d23d64b7d6addbeff04ac3496dd357f/R/SSplotComparisons.R#L738

The above `matplot` will generate a warning message when the entire matrix of points to move have NA values. I was going to implement an if statement around it such that the matplot is only called
```
if(!all(is.na(SpawnBio2[-(1:2), models])))
```
but I realized that there are multiple matplots in this function that would in theory generate the same warning message. I am not sure how to best combat the warning and thought maybe some good suggestions would be generated by bringing the issue up here rather than just fixing it "my way". Thanks! 

Idea 1 - implement "jittering" for all points but only move them if they are equal within a tolerance rather than having two separate matrices, i.e., adjust the value in a copy of the original matrix rather than plotting those that aren't moved and then plotting another matrix with the moved values?

**Comment by iantaylor-NOAA** [comment]
Thanks @kellijohnson-NOAA for bringing this up. Let me make sure I'm understanding the issue and we can brainstorm whether there's larger scale changes needed or a quick fix. The line you reference is related to adding points on top of the lines in the comparison plots as shown in the following figure.
![profile_M_compare1_spawnbio](https://user-images.githubusercontent.com/4992918/74060704-65840500-499f-11ea-9cc1-9f66fa823808.png)
Those points serve a few purposes:
- they help if the figure is printed in black and white (rare these days), 
- they help distinguish similar colors (including those that look similar to color-blind readers), and
- they help when one line overlaps another, like the red "Base Model" line associated with 0.449 covering up the lime green M = 0.45 line in the figure above.

The way those points get added is to copy the matrix of time series values for each model and use modular arithmetic to change the y-value to NA for all those that don't match the requirement
```
(SpawnBio2$Yr-initpoint)%%spacepoints != (staggerpoints*iline)%%spacepoints
```
where `initpoint`, `spacepoints`, and `staggerpoints` are all user-controlled inputs. I rarely change those from the defaults, but modifications I make are things like `spacepoints = 3000` to remove all the points, or fiddling with with `initpoint` and `spacepoints` to make the final point in each line of a retrospective analysis line up with the final year of data. The first of those changes could be replaced with some jittering algortithm, but the second couldn't. I might be the only one who ever fine tunes these settings, so for most users it would be sufficient to have 2 options: points that work vs. no points.

After writing all that, I finally tried to replicate the error and I now see the message
```
Warning messages:
1: In min(x) : no non-missing arguments to min; returning Inf
2: In max(x) : no non-missing arguments to max; returning -Inf
```
which appears to be some change in how `matplot` works (or I just forgot that it throws this error). That function is calculating a `ylim` value even if you have `add = TRUE` which means the value never gets used. I just provided a `ylim` input for those function calls in commit ffd9edcb86f4b1291e94d9fa8624b96ffb77b120 which seems to fix the issue. Let me know if that doesn't work for you and/or if you see further need for change.
